Sources/SKTestSupport/TestServer.swift

```diff
@@ -63,24 +64,23 @@ public struct TestSourceKitServer {
         let clientToServer: Pipe = Pipe()
         let serverToClient: Pipe = Pipe()
 
-        // FIXME: DispatchIO doesn't like when the Pipes close behind its back even after the tests
```

Comment:
fixed now, this is not DispatchIO's fault: In UNIX, closed file descriptors get re-used all the time (most syscalls guarantee to re-use the _lowest available_ number). If something closes a file descriptor that something else still owns this can be lead to anything from a crash to a security vulnerability. Imagine you close a file descriptor that DispatchIO wants to write to, if you're lucky the file descriptor is still closed but if you're unlucky, this has been reopened with a file descriptor to some file or maybe a network connection to somebody else. That would lead to that data being written to an unintended place (file or random network connection).

It's a terrible idea of `Foundation.Pipe` to close file descriptors on `deinit` but we can fix it with a simple `dup` of the file descriptors. So we just eagerly take ownership and close the one that `Pipe` created. That way everybody is happy.